File: frontend/src/app/services/polizza.service.ts
import { Injectable } from '@angular/core';
import { Polizza } from '../models/polizza.model';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PolizzaService {

  private baseUrl = '/Polizza';
  
  constructor(private http: HttpClient) {}

  getAll(): Observable<Polizza[]> {
    return this.http.get<Polizza[]>(`${this.baseUrl}/getPolizze`);
  }


  getById(id: number): Observable<Polizza> {
    return this.http.get<Polizza>(`${this.baseUrl}/getPolizzaById/${id}`);
  }
  
  
  getByClienteId(id: number): Observable<Polizza> {
    return this.http.get<Polizza>(`${this.baseUrl}/getPolizzaByClienteId/${id}`);
  }


  create(g: Polizza): Observable<Polizza> {
    return this.http.post<Polizza>(`${this.baseUrl}/insPolizza`, g);
  }


  update(g: Polizza): Observable<Polizza> {
    return this.http.put<Polizza>(`${this.baseUrl}/updPolizza`, g);
  }

  
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/deletePolizzaById/${id}`);
  }
  
}

File: backend/flexikmservlet/src/main/java/alexpre/flexikm/servlet/service/PolizzaService.java
package alexpre.flexikm.servlet.service;

import java.util.List;

import org.springframework.stereotype.Service;

import alexpre.flexikm.servlet.model.Polizza;

@Service
public interface PolizzaService
{
  public Polizza insert(Polizza polizza);

  public List<Polizza> getPolizze();

  public Polizza update(Polizza polizza);

  public String deletePolizzaById(Long id);

  public List<Polizza> getPolizzaById(Long id);

  public List<Polizza> getPolizzeByClienteId(Long id);
}
